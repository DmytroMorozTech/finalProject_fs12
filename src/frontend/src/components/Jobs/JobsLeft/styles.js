import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({

  container: {
    border: theme.border.simple,
    borderRadius: theme.shape.medium,
    backgroundColor: theme.palette.common.white,
    paddingBottom: theme.spacing(1),
    paddingTop: theme.spacing(1)
  }
}))
