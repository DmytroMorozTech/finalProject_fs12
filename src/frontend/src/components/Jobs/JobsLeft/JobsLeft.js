import styles from './styles'
import React from 'react'
import BookmarkIcon from '@material-ui/icons/Bookmark'
import NotificationsRoundedIcon from '@material-ui/icons/NotificationsRounded'
import LocalAtmIcon from '@material-ui/icons/LocalAtm'
import AssignmentTurnedInOutlinedIcon from '@material-ui/icons/AssignmentTurnedInOutlined'
import NoteIcon from '@material-ui/icons/Note'
import InsertDriveFileOutlinedIcon from '@material-ui/icons/InsertDriveFileOutlined'
import SettingsIcon from '@material-ui/icons/Settings'
import Item from './Item/Item'

function JobsLeft () {
  const classes = styles()

  const items = [
    { Icon: <BookmarkIcon fontSize="inherit"/>, title: 'My Jobs', onClick: () => console.log('My Jobs') },
    { Icon: <NotificationsRoundedIcon fontSize="inherit"/>, title: 'Job Alerts', onClick: () => console.log('Job Alerts') },
    { Icon: <LocalAtmIcon fontSize="inherit"/>, title: 'Salary', onClick: () => console.log('Salary') },
    { Icon: <AssignmentTurnedInOutlinedIcon fontSize="inherit"/>, title: 'Skill Assessments', onClick: () => console.log('Skill Assessments') },
    { Icon: <NoteIcon fontSize="inherit"/>, title: 'Interview Prep', onClick: () => console.log('Interview Prep') },
    { Icon: <InsertDriveFileOutlinedIcon fontSize="inherit"/>, title: 'Resume Builder', onClick: () => console.log('Resume Builder') },
    { Icon: <SettingsIcon fontSize="inherit"/>, title: 'Application Settings', onClick: () => console.log('Application Settings') }
  ]

  return (
    <div className={classes.container}>
      {items.map(({ Icon, title, onClick }, i) => (
        <Item key={i} Icon={Icon} title={title} onClick={onClick}/>
      ))}
    </div>
  )
}

export default JobsLeft
