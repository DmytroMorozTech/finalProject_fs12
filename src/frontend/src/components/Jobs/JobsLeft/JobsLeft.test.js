import { shallow } from 'enzyme'
import React from 'react'
import JobsLeft from './JobsLeft'

jest.mock('./styles', () => () => ({}))

const setUp = () => shallow(<JobsLeft />)

describe('should render JobsLeft component', () => {
  let component
  beforeEach(() => {
    component = setUp()
  })
  afterEach(() => {
    jest.restoreAllMocks()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
