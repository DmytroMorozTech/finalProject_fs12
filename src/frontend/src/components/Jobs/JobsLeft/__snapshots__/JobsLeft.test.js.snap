// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render JobsLeft component to match snapshot 1`] = `
<div>
  <JobsLeftItem
    Icon={
      <Memo
        fontSize="inherit"
      />
    }
    key="0"
    title="My Jobs"
    to="/jobs/my_jobs"
  />
  <JobsLeftItem
    Icon={
      <Memo
        fontSize="inherit"
      />
    }
    key="1"
    title="Job Alerts"
    to="#"
  />
  <JobsLeftItem
    Icon={
      <Memo
        fontSize="inherit"
      />
    }
    key="2"
    title="Skill Assessments"
    to="#"
  />
  <JobsLeftItem
    Icon={
      <Memo
        fontSize="inherit"
      />
    }
    key="3"
    title="Interview Prep"
    to="#"
  />
  <JobsLeftItem
    Icon={
      <Memo
        fontSize="inherit"
      />
    }
    key="4"
    title="Resume Builder"
    to="#"
  />
  <JobsLeftItem
    Icon={
      <Memo
        fontSize="inherit"
      />
    }
    key="5"
    title="Application Settings"
    to="#"
  />
</div>
`;
